pkg/cluster/internal/delete/delete.go

```diff
@@ -44,6 +44,8 @@ func Cluster(logger log.Logger, p providers.Provider, name, explicitKubeconfigPa
 			return err
 		}
 		logger.V(0).Infof("Deleted nodes: %q", n)
+	} else {
+		logger.V(0).Infof("Whooops, no nodes found for cluster %q", name)
```

Comment:
The idempotent nature of cluster deletion has been brought up several times in the past. We do not want to do this. If you need to know if a cluster exists or not before deleting, then you should list the clusters first.

The line above was added as a compromise. So another option is to check the output an make sure it contains the string "Deleted nodes" if you want to verify something was actually deleted.

But functionality-wise, if you tell kind "I don't want a cluster named X", at the end you will not have a cluster named X. Whether there was one to begin with or not.

See also the [note in the docs](https://kind.sigs.k8s.io/docs/user/quick-start/#deleting-a-cluster) explicitly stating this expected behavior.

/hold

Reply:
Right, the idempotent nature is important to reliable CI because it makes it safe and even encouraged to have both the normal cleanup path and a signal / exit handler for cleanup without worrying about coordinating them. You can just cleanup on exit always and it's fine and reliable.

logging warnings like this will discourage implementing reliable cleanup and the output is misleading anyhow (a node or nodes is not the sum of a kind cluster)